test/test_cli.py

```diff
@@ -21,35 +21,40 @@
 from pyscitt.governance import ProposalNotAccepted
 
 
-def run(*cmd, **kwargs):
-    args = [str(c) for c in cmd]
-    for k, v in kwargs.items():
-        flag = k.replace("_", "-")
-        if v is True:
-            args += [f"--{flag}"]
-        elif v is False:
-            pass
-        elif isinstance(v, (str, Path)):
-            args += [f"--{flag}={v}"]
-        else:
-            raise TypeError(f"Invalid value: {v}")
-
-    LOG.info(shlex.join(["scitt"] + args))
-    main(args)
-
-
-def test_smoke_test(tmp_path: Path):
-    url = os.environ.get("CCF_URL", "https://127.0.0.1:8000")
-    server_args = {
-        "url": url,
-        "development": True,
-    }
+@pytest.fixture
+def run(request):
+    def f(*cmd, _with_service_url=False, _with_member_auth=False, **kwargs):
+        if _with_service_url:
+            kwargs["url"] = request.getfixturevalue("service_url")
+            kwargs["development"] = True
+
+        if _with_member_auth:
+            paths = request.getfixturevalue("member_auth_path")
+            kwargs["member_cert"] = paths[0]
+            kwargs["member_key"] = paths[1]
+
+        args = [str(c) for c in cmd]
+        for k, v in kwargs.items():
+            flag = k.replace("_", "-")
+            if v is True:
+                args += [f"--{flag}"]
+            elif v is False:
+                pass
+            elif isinstance(v, (str, Path)):
+                args += [f"--{flag}={v}"]
+            else:
+                raise TypeError(f"Invalid value: {v}")
+
+        LOG.info(shlex.join(["scitt"] + args))
+        main(args)
+
+    return f
+
 
+def test_smoke_test(run, client, tmp_path: Path):
     trust_store = tmp_path / "store"
     trust_store.mkdir()
-    (trust_store / "service.json").write_text(
-        json.dumps(Client(**server_args).get_parameters())
-    )
+    (trust_store / "service.json").write_text(json.dumps(client.get_parameters()))
 
     (tmp_path / "claims.json").write_text(json.dumps({"foo": "bar"}))
 
```

Comment:
I actually want to test the `propose_ca_certs` command which is why I'm using the class directly instead of the fixture, as the latter does the CA cert stuff automatically.

In general I think this module should avoid using the python APIs as much as possible (yeah the `client.get_parameters()` call above is not great, but also making a CLI for just this seems overkill, maybe).

Reply:
I'll add a comment to explain this, similar to `test_default_did_port`.